### PR TITLE
WIP: GNOME 45

### PR DESCRIPTION
**Pull request by paper42**
very WIP, probably not functional yet

#### Testing the changes
- I tested the changes in this PR: **NO**

depends on:
* #46530
* #46529
* #45570 
* #46507
* #46532
* #46622
* #46623

[ci skip]

### CODE REVIEW

**Comment by let-the-world-forget-the-flower** [comment]
loupe and snapshot should be added as they are the new core apps (and replace eog, cheese)

**Comment by chrysos349** [comment]
@let-the-world-forget-the-flower

`snapshot` requires `pango>=1.51`
`loupe` requires `glycin` which requires `cairo>=1.17` ( #46356 needs to be merged first)

File: srcpkgs/mutter/template [hunk 1]
```diff
@@ -12,18 +12,19 @@ hostmakedepends="gettext glib-devel pkg-config zenity wayland-devel xorg-server
 makedepends="elogind-devel glib-devel gnome-desktop-devel graphene-devel
  json-glib-devel libglib-devel libSM-devel libXtst-devel libcanberra-devel
  libinput-devel MesaLib-devel pipewire-devel startup-notification-devel gtk4-devel
- wayland-protocols gnome-settings-daemon-devel libgudev-devel libwacom-devel"
+ wayland-protocols gnome-settings-daemon-devel libgudev-devel libwacom-devel libei"
```
Comment (averyterrel) [inline srcpkgs/mutter/template:15]: `libei` should be `libei-devel`

**Comment by corigan01** [comment]
I was able to build with the following changes. Hopefully they help someone. 
- gnome-connections required `fuse3-devel` to build. 
- Mutter required averyterrel's change to `libei-devel`
- I updated Pango to 1.51.0


srcpkgs/gnome-connections/template
```diff
 makedepends="gtk+3-devel libhandy1-devel gtk-vnc-devel libgcrypt-devel
- gnutls-devel libsasl-devel libsecret-devel freerdp-devel"
+ gnutls-devel libsasl-devel libsecret-devel freerdp-devel fuse3-devel"
```

srcpkgs/mutter/template
```diff
 makedepends="elogind-devel glib-devel gnome-desktop-devel graphene-devel
  json-glib-devel libglib-devel libSM-devel libXtst-devel libcanberra-devel
  libinput-devel MesaLib-devel pipewire-devel startup-notification-devel gtk4-devel
- wayland-protocols gnome-settings-daemon-devel libgudev-devel libwacom-devel libei"
+ wayland-protocols gnome-settings-daemon-devel libgudev-devel libwacom-devel libei-devel"
```

srcpkgs/pango/template
```diff
 # Template file for 'pango'
 pkgname=pango
-version=1.50.14
+version=1.51.0
 revision=1
 build_style=meson
 build_helper=gir
@@ -14,7 +14,7 @@ license="LGPL-2.1-or-later"
 homepage="https://www.pango.org/"
 changelog="https://gitlab.gnome.org/GNOME/pango/-/raw/main/NEWS"
 distfiles="${GNOME_SITE}/pango/${version%.*}/pango-${version}.tar.xz"
-checksum=1d67f205bfc318c27a29cfdfb6828568df566795df0cb51d2189cde7f2d581e8
+checksum=74efc109ae6f903bbe6af77eaa2ac6094b8ee245a2e23f132a7a8f0862d1a9f5
 make_check=no  # doesn't pass its own tests
```

**Comment by paper42** [comment]
Thanks a lot for the work, would you mind submitting your changes to my fork pointing to this branch? (paper42:gnome45) so it doesn't get lost?

**Comment by corigan01** [comment]
> Thanks a lot for the work, would you mind submitting your changes to my fork pointing to this branch? (paper42:gnome45) so it doesn't get lost? I am not so sure about the pango and zenity updates since they not stable and we will need a mozjs and webkit2gtk update for full GNOME 45.

I submitted as a Pull-Request to your `gnome45` branch. Hope that is okay!

**Comment by LokiRazer** [comment]
Can someone merge this PR please? Or is there any work to do? The requested PRs are also merged

**Comment by paper42** [comment]
I don't work on Void anymore, if anyone wants to pick this up, feel free to do so and take this PR as a base and add https://github.com/paper42/void-packages/pull/8.

**Comment by Jaymz665** [comment]
> I don't work on Void anymore, if anyone wants to pick this up, feel free to do so and take this PR as a base and add [paper42#8](https://github.com/paper42/void-packages/pull/8).

Well, let's sit and wait, if there is no movement for a month, I will return to the distrohope.
Gnome 46 is already looming on the threshold. I love void as a stable rolling, but I love Gnome more

**Comment by fvalasiad** [comment]
@paper42 I haven't worked on anything other than basic standalone packages, I would be willing to allocate some of my time to get this PR merged, but I'd like some pointers on what seems to be the issue so far, what qualities should the package satisfy? Am I expected to thoroughly test gnome as a whole before I open the PR?

**Comment by Jaymz665** [comment]
> Well, let's sit and wait, if there is no movement for a month, I will return to the distrohope. Gnome 46 is already looming on the threshold. I love void as a stable rolling, but I love Gnome more

I deceived myself. I liked void too much. But I need a fresh gnome....

**Comment by fvalasiad** [comment]
@Jaymz665 it's unfortunate but it is what it is, quite surprising that no one of the contributors daily drives gnome or at least cares enough to keep it up to date.

The gnome distros outnumber the rest by a margin though, if I were you I'd aim at fedora, best of luck.

**Comment by Jaymz665** [comment]
@fvalasiad 
изображение

If you want to do something, do it yourself =)

**Comment by corigan01** [comment]
I have been using my gnome 45 since I made the comment before. Seems pretty stable, haven't had any issues.

**Comment by fvalasiad** [comment]
PR when? @corigan01 @Jaymz665

**Comment by Jaymz665** [comment]
> PR when? 

I'll try to make my own branch, I won't talk about mr (idk how), but I'll try to keep it up to date =)

**Comment by Jaymz665** [comment]
I'm not sure yet that I did everything right (I'm not really a git user) but here: https://github.com/Jaymz665/void-packages/tree/Gnome-45.4

**Comment by zuranota** [comment]
> I'm not sure yet that I did everything right (I'm not really a git user) but here: https://github.com/Jaymz665/void-packages/tree/Gnome-45.4

Bro i am not a tech savvy nor i understand git much, but will this branch get merged with the official repo? or i have to build it from source? Anyways thank you for taking the initiative.

**Comment by Jaymz665** [comment]
You can build from the source code, or use ready-made binary files.
So far, I've tried it with the members of the telegram group and it works.
If you already have void-src, then it is enough to download the archive from here:
https://github.com/Jaymz665/void-packages/tree/Gnome-45.4/binpkgs
Place it in the "hostdir` folder and run the command
"`
sudo xbps-install --repository hostdir/binpkgs gnome
```
This worked for all participants with x64 glibc. I am still learning these things, I hope I will develop it further

**Comment by zuranota** [comment]
@Jaymz665  It seems easy enough. But will it ever get merged to official repo? just curious. Thank you again.

**Comment by fvalasiad** [comment]
@zuranota If anyone bothers to pick it up and make a proper PR, yes.

**Comment by fanyx** [comment]
i can pick up paper's branch, ads jaymz' commits on top and see if that works in the void pipelines

have no clue how to troubleshoot though, if any issues arise
atleast i could do the git rebasing ^^

**Comment by Jaymz665** [comment]
@fanyx In general, you can take the entire branch, it is based on the base branch, I transferred the gnome files from paper and updated them

**Comment by zuranota** [comment]
@Jaymz665  Tried to install it but it is giving me this error:
`MISSING: nautilus>=45.2
Transaction aborted due to unresolved dependencies.`
I followed your instructions.

**Comment by Jaymz665** [comment]
@zuranota
Oh, my fault, I completely forgot. Take this for now:
https://github.com/void-linux/void-packages/pull/47121

I'll fix it one of these days

**Comment by GGTNT** [comment]
hey I did everything as you said but I have a mistake :/ 
```
eog-45.0_1: verifying SHA256 hash...
ERROR: eog-45.0_1: SHA256 hash is not valid: No such file or folder
Transaction failed! see above for errors.
```
And it's the same for all packages, any idea where it's coming from?

**Comment by Jaymz665** [comment]
@GGTNT  Let's make it easier, create a folder, unzip the archive there

https://drive.google.com/file/d/1iBFX4VsvNtUNy47Euh-XsDJPEIp3QTg6/view?usp=sharing

and run the commands from the created folder:
```
sudo xbps-install --repository hostdir/binpkgs nautilus
sudo xbps-install --repository hostdir/binpkgs gnome
```

**Comment by GGTNT** [comment]
@Jaymz665 Okay, that works for me, thanks

**Comment by shnaps** [comment]
Just wait a couple of days, PR for gnome 45 is on the way https://github.com/void-linux/void-packages/pull/48762

**Pull request closed without merging**